scheduler/src/cook/mesos/scheduler.clj

```diff
@@ -1059,25 +1058,23 @@
 
    A lingering task is a task that runs longer than timeout-hours."
   [db now max-timeout-hours default-timeout-hours]
-  (->> (q '[:find ?task-id ?start-time ?max-runtime
-            :in $ ?default-runtime
-            :where
-            [(ground [:instance.status/unknown :instance.status/running]) [?status ...]]
-            [?i :instance/status ?status]
-            [?i :instance/task-id ?task-id]
-            [?i :instance/start-time ?start-time]
-            [?j :job/instance ?i]
-            [(get-else $ ?j :job/max-runtime ?default-runtime) ?max-runtime]]
-          db (-> default-timeout-hours time/hours time/in-millis))
-       (keep (fn [[task-id start-time max-runtime]]
-               ;; The convertion between random time units is because time doesn't like
-               ;; Long and Integer/MAX_VALUE is too small for milliseconds
-               (let [timeout-minutes (min (.toMinutes TimeUnit/MILLISECONDS max-runtime)
-                                          (.toMinutes TimeUnit/HOURS max-timeout-hours))]
```

Comment:
@wyegelwel - This code was rounding any sub-minute values of `:max-runtime` to zero minutes, which was causing the `test_max_runtime_exceeded` integration test to fail nondeterministically. I don't understand the limitation described in the comment above, so I wanted to double-check with you to make sure that by switching to milliseconds I'm not breaking something that's not currently covered by our tests.

Reply:
@shamsimam - Wil and I had a chat and resolved this. See the comment on `millis->period` below for more details.